lua/tint/colors.lua

```diff
@@ -0,0 +1,83 @@
+local colors = {}
+
+--- Format a decimal as a 6-digit hex value
+---
+---@param val number Number to format as 6-digit hex
+---@return string The color represented as 6-digit hex
+colors.get_hex = function(val)
+  return string.format("%06x", val)
+end
+
+--- Clamp a value between 0 and 255
+---
+---@param component number A number to bound within [0, 255]
+---@return string The color clamped between 0 and 255
+colors.clamp = function(component)
+  return math.min(math.max(component, 0), 255)
+end
+
+--- Generate function to saturate a color by the specified amount
+---
+---@param amt number The amount to saturate the color by
+colors.saturate = function(amt)
+  return function(r, g, b, _)
+    if amt ~= 1 then
+      local rec601_luma = 0.299 * r + 0.587 * g + 0.114 * b
+
+      r = math.floor(r * amt + rec601_luma * (1 - amt))
+      g = math.floor(g * amt + rec601_luma * (1 - amt))
+      b = math.floor(b * amt + rec601_luma * (1 - amt))
+    end
+
+    return r, g, b
+  end
+end
+
+--- Generate function to lighten or darken a color by the specified amount
+---
+---@param amt number The amount to lighten or darken the color by
+colors.tint = function(amt)
+  return function(r, g, b, _)
+    return r + amt, g + amt, b + amt
+  end
+end
+
+--- Transform a color given a change in tint and saturation
+---
+---@param hex string The hex color to transform.
+---@param transforms table A table of functions used to transform the input hex color
+---@return string The hex representation color transformed by the configured values
+colors.transform_color = function(hl_group_info, hex, transforms)
```

Comment:
Seems like `hl_group_info` is missing a `@param` definition (pointing this out because it's not immediately clear to me what it is, so I was looking for it's explanation)

Reply:
Fixed.